packages/dev/core/src/Meshes/abstractMesh.hotSpot.ts

```diff
@@ -0,0 +1,119 @@
+import { Vector2, Vector3, TmpVectors, Matrix } from "../Maths/math.vector";
+import type { AbstractMesh } from "./abstractMesh";
+import { VertexBuffer } from "../Buffers/buffer";
+
+/**
+ * Data for mesh hotspot computation
+ */
+export class HotSpotQuery {

```

Comment:
I think we should make this a type instead of a class. It is all public, and has no functions, and there are benefits to using a type instead of a class. For example, you can do things like `export const myHotSpotQuery = { pointIndex: [1,2,3], barycentric: [4,5,6] } as const` which is side effect free, whereas `const myHotSpotQuery = new HotSpotQuery(); myHotSpotQuery.pointIndex = [1,2,3]; export myHotSpotQuery` is considered to have side effects (since the constructor executes JS code that could do anything).